regolith/builders/recentcollabsbuilder.py

```diff
@@ -112,26 +134,49 @@ def latex(self):
-                                        "begin_year": 2019}])
+                                            "begin_year": 2019}])
                 emp.sort(key=ene_date_key, reverse=True)
-                people_df = pd.DataFrame(ppl_names)
-                out_folder = "_build/recent-collabs/"
-                out_file = 'recent_collaborators.csv'
-                people_df.to_csv(''.join([out_folder, out_file], index=False) #save csv file
-                coa_excel_file = os.path.join(os.path.dirname(os.path.dirname(__file__)),
-                                              "templates", "coa_template.xlsx"
-                )
-                excel_df = pd.read_excel(coa_excel_file, header=None)
-                head_part = excel_df.iloc[:COAUTHOR_TABLE_OFFFSET, :].copy()
-                tail_part = excel_df.iloc[COAUTHOR_TABLE_OFFFSET:, :].copy()
-                concat_df = pd.concat([head_part, people_df, tail_part], ignore_index=True)
-                concat_df.to_excel(''.join([out_folder, 'coa_table.xlsx']), header=None, index=False)
-
+        return ppl_names
 
+    def make_csv_and_excel(self, id, months):
+        '''
+        function to fill in the 'coa_template.xlsx' and make a csv file with the people and institutions
+        information, output from self.get_ppl_inst_info(id, months)
+        args: see self.get_ppl_inst_info(id, months) for details of the args
+        output: None, save an excel file "coa_tables.xlsx" and csv 'recent_collaborators.csv'
+        in _build/recent-collabs/
+        '''
+        ppl_names = self.get_ppl_inst_info(id, months)
+        # make csv
+        ppl_df = pd.DataFrame(ppl_names)
+        ppl_df.columns = ['', 'Name', 'Institution', 'Optional Info (Email/Department)', 'Last Active']
+        out_folder = "_build/recent-collabs/"
+        out_file = "recent_collaborators.csv"
+        ppl_df.to_csv(''.join([out_folder, out_file]), index=False)
+        # fill in excel
+        coa_excel_file = os.path.join(os.path.dirname(os.path.dirname(__file__)),
+                                      "templates", "coa_template.xlsx")
+        # loading excel file
+        wb = openpyxl.load_workbook(coa_excel_file)
+        ws = wb.worksheets[0]
+        num_rows = len(ppl_names)  # number of rows to be added to the excel
+        num_colns = len(ppl_names[0])  # number of columns
+        # add empty rows below the header
+        ws.insert_rows(COAUTHOR_TABLE_OFFFSET + 1, num_rows)
+        # openpyxl index column and row from 1 instead
+        # filling in the info
+        for row_idx in range(1, num_rows + 1):
+            for col_idx in range(1, num_colns + 1):
+                cell = ws.cell(row=row_idx + COAUTHOR_TABLE_OFFFSET, column=col_idx)
+                cell.value = ppl_names[row_idx-1][col_idx - 1]
+                cell.font = font
+                cell.border = border
+        wb.save(''.join([out_folder, 'coa_table.xlsx']))
 
+    def latex(self):
```

Comment:
What is this function for?  It seems to have things hard-coded. Is it used for testing?  If so, please remove as testing should be handled by the testing framework.

Reply:
The 'latex' is the function that will be called to make the files. We will reformat the design like this:

'latex' do the taks by two steps: first, query the data from the database (which is a list of name and institution); second, use the data to generate csv and xlsx. Each step is done by a function (a 'get_data' function and a 'use_data' function).

Reply:
OK, I like that approach.  So we just need to remove the hard-coded sbillinge and 48.

The default behavior would normally be to loop over all people in people.yml, but override this behavior with the people list if one is specified at the command line.

for the 48, I would create a constant global variable at the top of the module with a helpful name, like ``LINES_TO_SKIP_TO_COLLABORATORS_TABLE_IN_EXCEL_FILE = 48`` and then put the constant name here.